client/common/configuration/services/channel-details.ts

```diff
@@ -0,0 +1,161 @@
+    const validationError: ValidationError = {
+      name: 'ChannelConfigurationOptionsValidationError',
+      message: 'There was an error during validation',
+      properties: {}
+    }
+    const botConf = channelConfigurationOptions.bot
+    const slowModeDuration = channelConfigurationOptions.slowMode.duration
+
+    validationError.properties['slowMode.duration'] = []
+
+    if (
+      (typeof slowModeDuration !== 'number') ||
+      isNaN(slowModeDuration)) {
+      validationError.properties['slowMode.duration'].push(ValidationErrorType.WrongType)
+      hasErrors = true
+    } else if (
+      slowModeDuration < 0 ||
+      slowModeDuration > 1000
+    ) {
+      validationError.properties['slowMode.duration'].push(ValidationErrorType.NotInRange)
+      hasErrors = true
+    }
+
+    // If !bot.enabled, we don't have to validate these fields:
+    // The backend will ignore those values.
+    if (botConf.enabled) {
+      validationError.properties['bot.nickname'] = []
+
+      if (/[^\p{L}\p{N}\p{Z}_-]/u.test(botConf.nickname ?? '')) {
+        validationError.properties['bot.nickname'].push(ValidationErrorType.WrongFormat)
+        hasErrors = true
+      }
+
+      for (const [i, fw] of botConf.forbiddenWords.entries()) {
+        for (const v of fw.entries) {
+          validationError.properties[`bot.forbiddenWords.${i}.entries`] = []
+          if (fw.regexp) {
+            if (v.trim() !== '') {
+              try {
+                const test = new RegExp(v)
+                test.test(v)
+              } catch (_) {
+                validationError.properties[`bot.forbiddenWords.${i}.entries`]
+                  .push(ValidationErrorType.WrongFormat)
+                hasErrors = true
+              }
+            }
+          }
+        }
+      }
+
+      for (const [i, cd] of botConf.commands.entries()) {
+        validationError.properties[`bot.commands.${i}.command`] = []
+
+        if (/\s+/.test(cd.command)) {
+          validationError.properties[`bot.commands.${i}.command`].push(ValidationErrorType.WrongFormat)
+          hasErrors = true
+        }
+      }
+    }
+
+    if (hasErrors) {
+      throw validationError
+    }
+
+    return true
+  }
+
+  saveOptions = async (channelId: number,
+    channelConfigurationOptions: ChannelConfigurationOptions): Promise<Response> => {
+    if (!await this.validateOptions(channelConfigurationOptions)) {
+      throw new Error('Invalid form data')
+    }
+
+    const response = await fetch(
+      getBaseRoute(this._registerClientOptions) + '/api/configuration/channel/' + encodeURIComponent(channelId),
+      {
+        method: 'POST',
+        headers: this._headers,
+        body: JSON.stringify(channelConfigurationOptions)
+      }
+    )
+
+    if (!response.ok) {
+      throw new Error('Failed to save configuration options.')
+    }
+
+    return response.json()
+  }
+
+  fetchUserChannels = async (username: string): Promise<ChannelLiveChatInfos[]> => {
```

Comment:
Is this used?
Is it not related to configuration home?

Reply:
It is used and related to home, that's why this service has a more broader name.